--- com/univocity/articles/csvcomparison/parser/FlatpackParser.java
+++ com/univocity/articles/csvcomparison/parser/FlatpackParser.java
@@ -17,6 +17,7 @@ package com.univocity.articles.csvcomparison.parser;
 
 import java.io.*;
 import java.util.*;
+import java.util.Optional;
 
 import net.sf.flatpack.*;
 
@@ -42,13 +43,16 @@ class FlatpackParser extends AbstractParser {
 		DataSet dataset = parser.parse();
 
 		while (dataset.next()) {
-			Record record = dataset.getRecord();
-			String[] row = new String[record.getColumns().length];
-			int i = 0;
-			for (String column : record.getColumns()) {
-				row[i++] = record.getString(column);
+			final Optional<Record> maybeRecord = dataset.getRecord();
+			if(maybeRecord.isPresent()) {
+				final Record record = maybeRecord.get();
+				String[] row = new String[record.getColumns().length];
+				int i = 0;
+				for (String column : record.getColumns()) {
+					row[i++] = record.getString(column);
+				}
+				rows.add(row);
 			}
-			rows.add(row);
 		}
 		return rows;
 	}
